### PR TITLE
zh-translation: content/zh/docs/ops/common-problems/security-issues/i…

### PR DESCRIPTION
**Pull request by caoyukun0430**
[ ] Configuration Infrastructure
[X] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Review by SataQiu — COMMENTED** [review]
/ok-to-test

File: content/zh/docs/ops/common-problems/security-issues/index.md [hunk 1]
```diff
@@ -8,162 +8,186 @@ aliases:
...
-      principalBinding: USE_ORIGIN
+      selector:
+        matchLabels:
+          app: httpbin
+      jwtRules:
+      - issuer: "testing@secure.istio.io"
+        jwksUri: "{{< github_file >}}/security/tools/jwt/samples/jwks.json"
     {{< /text >}}
 
 1. 如果 JWT token 放在 http 请求头 Authorization 字段值中，需要确认 JWT token 的有效性（未过期等）。JWT 令牌中的字段可以使用在线 JWT 解析工具进行解码，例如：[jwt.io](https://jwt.io/)。
```
Comment (orangegzx) [inline content/zh/docs/ops/common-problems/security-issues/index.md:35]:
```suggestion
1. 如果 JWT Token 放在 HTTP 请求头 Authorization 字段值中，需要确认 JWT Token 的有效性（未过期等）。JWT 令牌中的字段可以使用在线 JWT 解析工具进行解码，例如：[jwt.io](https://jwt.io/)。
```

Comment (orangegzx) [reply content/zh/docs/ops/common-problems/security-issues/index.md:35]: 建议专业名词大写

Comment (caoyukun0430) [reply content/zh/docs/ops/common-problems/security-issues/index.md]: 谢谢，已修改

File: content/zh/docs/ops/common-problems/security-issues/index.md [hunk 2]
```diff
@@ -8,162 +8,186 @@ aliases:
...
+owner: istio/wg-security-maintainers
+test: no
 ---
 
 ## 终端用户认证失败{#end-user-authentication-fails}
 
-使用 Istio，可以启用终端用户认证。目前，Istio 认证策略提供的终端用户凭证是 JWT。以下是排查终端用户 JWT 身份认证问题的指南。
-
-1. 检查 Istio 身份认证策略 Policy 配置，`principalBinding` 需要被设为 `USE_ORIGIN` 来验证终端用户。
+使用 Istio，可以通过 [请求认证策略](/zh/docs/tasks/security/authentication/authn-policy/#end-user-authentication) 启用终端用户认证。目前，Istio 认证策略提供的终端用户凭证是 JWT。以下是排查终端用户 JWT 身份认证问题的指南。
```
Comment (MXuDong) [inline content/zh/docs/ops/common-problems/security-issues/index.md:17]:
```suggestion
使用 Istio，可以通过[请求认证策略](/zh/docs/tasks/security/authentication/authn-policy/#end-user-authentication) 启用终端用户认证。目前，Istio 认证策略提供的终端用户凭证是 JWT。以下是排查终端用户 JWT 身份认证问题的指南。
```

File: content/zh/docs/ops/common-problems/security-issues/index.md [hunk 3]
```diff
@@ -8,162 +8,186 @@ aliases:
...
+        - /foo
+  - from:
+    - source:
+        namespaces:
+        - foo
+{{< /text >}}
 
-1. 在 Kubernetes 环境，确保在一个 `ServiceRole` 对象下的所有服务都和 `ServiceRole` 在同一个 namespace 。例如，如果 `ServiceRole` 对象中的服务是 `a.default.svc.cluster.local`，`ServiceRole` 必须在 `default` 命名空间（`metadata/namespace` 这一行应该是 `default`）。对于非 Kubernetes 的环境，一个网格的所有 `ServiceRoles` 和 `ServiceRoleBindings` 都应该在相同的命名空间下。
+您期望的策略所允许的请求是符合路径为 `/foo` **and** 源命名空间为 `foo`。
+但是，策略实际上允许的请求是符合路径为 `/foo` **or** 源命名空间为 `foo`,这显然会更加宽松。
```
Comment (MXuDong) [inline content/zh/docs/ops/common-problems/security-issues/index.md:108]:
```suggestion
您期望的策略所允许的请求是符合路径为 `/foo` **and** 源命名空间为 `foo`。但是，策略实际上允许的请求是符合路径为 `/foo` **or** 源命名空间为 `foo`,这显然会更加宽松。
```

Comment (MXuDong) [inline content/zh/docs/ops/common-problems/security-issues/index.md:108]:
```suggestion
您期望的策略所允许的请求是符合路径为 `/foo` **and** 源命名空间为 `foo`。但是，策略实际上允许的请求是符合路径为 `/foo` **or** 源命名空间为 `foo`,这显然会更加宽松。
```

File: content/zh/docs/ops/common-problems/security-issues/index.md [hunk 4]
```diff
@@ -8,162 +8,186 @@ aliases:
...
+在 YAML 的语义中，`from:` 前面的 `-` 意味着这是列表中的一个新元素。这会在策略中创建两条规则，而不是所希望的一条。在认证策略中，多条规则之间是 `OR` 的关系。
 
-## 授权太过宽松{#authorization-is-too-permissive}
+为了解决这个问题，只需要将多余的 `-` 移除，这样策略就只有一条规则来允许符合路径为 `/foo` **and** 源命名空间为 `foo` 的请求，这样就更加严格了。
 
-如果已经对一个服务启用了授权，但是对这个服务的请求没有被阻止，那么很有可能是授权没有成功启用。通过以下步骤可以检查这种情况：
+### 确保您没有在 TCP 端口上使用仅适用于 HTTP 的字段{#make-sure-you-are-not-using-http-only-fields-on-tcp-ports}
 
-1. 检查[启用授权文档](/zh/docs/concepts/security/#authorization)来正确的启用 Istio 授权。
+授权策略会变得更加严格因为定义了仅适用于 HTTP 的字段 (比如 `host`, `path`, `headers`, JWT， 等等) 在纯 TCP 连接上使不存在的。
```
Comment (MXuDong) [inline content/zh/docs/ops/common-problems/security-issues/index.md:116]:
```suggestion
授权策略会变得更加严格因为定义了仅适用于 HTTP 的字段 (比如 `host`，`path`，`headers`，JWT， 等等) 在纯 TCP 连接上是不存在的。
```

File: content/zh/docs/ops/common-problems/security-issues/index.md [hunk 5]
```diff
@@ -8,162 +8,186 @@ aliases:
...
 
-1. 避免为 Istio 控制面组件启用授权，包括 Mixer，Pilot 和 Ingress。Istio 的授权是设计用于 Istio Mesh 下的服务之间的授权的。对 Istio 组件启用授权会引发不可预料的行为。
+对于 `ALLOW` 类的策略来说，这些字段不会被匹配。但对于 `DENY` 以及 `CUSTOM` 类策略来说，这类字段会被认为是始终匹配的。最终结果会是一个更加严格的策略从而可能导致意外的连接拒绝。
 
-1. 在你的 Kubernetes 环境中，检查所有命名空间下的部署，确保没有可能导致 Istio 错误的遗留的部署。如果发现在向 Envoy 推送授权策略的时候发生错误，你可以禁用 Pilot 的授权插件。
+检查 Kubernetes 服务定义来确定端口是 [命名中包含正确的协议名称](/zh/docs/ops/configuration/traffic-management/protocol-selection/#manual-protocol-selection)。如果您在端口上使用了仅适用于 HTTP 的字段，要确保端口名有 `http-` 前缀。
```
Comment (MXuDong) [inline content/zh/docs/ops/common-problems/security-issues/index.md:120]:
```suggestion
检查 Kubernetes 服务定义来确定端口是[命名中包含正确的协议名称](/zh/docs/ops/configuration/traffic-management/protocol-selection/#manual-protocol-selection)。如果您在端口上使用了仅适用于 HTTP 的字段，要确保端口名有 `http-` 前缀。
```

File: content/zh/docs/ops/common-problems/security-issues/index.md [hunk 6]
```diff
@@ -8,162 +8,186 @@ aliases:
...
+- 当一个工作负载上同时配置了多个动作时 (`CUSTOM`, `ALLOW` 和 `DENY`)， 所有的动作必须都满足。换句话说，如果有任何一个动作拒绝该请求，那么该请求会被拒绝，并且只有所有的动作都允许了该请求，该请求才会被允许。
 
-1. 这里应该**只有一个**命名为 `default` 的 `ClusterRbacConfig` 实例。否则 Istio 会禁用授权功能并忽略所有策略。
+- 在任何情况下，`AUDIT` 动作不会实施控制访问权并且不会拒绝请求。
 
-    {{< text plain >}}
-    NAMESPACE   NAME      AGE
-    default     default   1d
-    {{< /text >}}
+## 确保 Istiod 接受策略{#ensure-pilot-accepts-the-policies}
```
Comment (MXuDong) [inline content/zh/docs/ops/common-problems/security-issues/index.md:134]:
```suggestion
## 确保 Pilot 接受策略{#ensure-pilot-accepts-the-policies}
```
我认为这里应该保留 Pilot 而不是 Istiod，不然和锚点对不上，如果语义确实应该是 Istiod 的话，就算了

File: content/zh/docs/ops/common-problems/security-issues/index.md [hunk 7]
```diff
@@ -8,162 +8,186 @@ aliases:
...
+    说明 Istiod 生成了：
 
-    - `sleep.foo.svc.cluster.local` 的空配置。因为没有符合条件的策略，并且 Istio 默认情况下，会禁止所有对这一服务的访问。
+    - 对于带 `app=ext-authz-server,...` 标签的负载生成了带有 `ns[foo]-policy[ext-authz-server]-rule[0]` 策略的 HTTP 过滤器配置。
 
-    - `productpage.default.svc.cluster.local` 的配置。Istio 会放行所有针对该服务的 GET 访问。
+    - 对于带 `app=ext-authz-server,...` 标签的负载生成了带有 `ns[foo]-policy[ext-authz-server]-rule[0]` 策略的 TCP 过滤器配置。
 
-## 确认 Pilot 正确的将策略分发给了代理服务器{#ensure-pilot-distributes-policies-to-proxies-correctly}
+## 确认 Istiod 正确的将策略分发给了代理服务器{#ensure-pilot-distributes-policies-to-proxies-correctly}
```
Comment (MXuDong) [inline content/zh/docs/ops/common-problems/security-issues/index.md:190]:
```suggestion
## 确认 Istiod 正确的将策略分发给了代理服务器{#ensure-istiod-distributes-policies-to-proxies-correctly}
```

File: content/zh/docs/ops/common-problems/security-issues/index.md [hunk 8]
```diff
@@ -317,41 +341,19 @@ Pilot 负责向代理服务器分发授权策略。下面的步骤用来确认 P
 
 ## 密钥和证书错误{#keys-and-certificates-errors}
 
-如果您怀疑 Istio 使用的某些密钥或证书不正确，那么第一步是确保 [Citadel 健康](#repairing-citadel)。
-
-然后，您可以验证 Citadel 是否真正生成了密钥和证书：
+如果您怀疑 Istio 使用的某些密钥或证书不正确，您可以检查任何 pod 的内容信息。
```
Comment (MXuDong) [inline content/zh/docs/ops/common-problems/security-issues/index.md:344]:
```suggestion
如果您怀疑 Istio 使用的某些密钥或证书不正确，您可以检查任何 Pod 的内容信息。
```

Comment (caoyukun0430) [reply content/zh/docs/ops/common-problems/security-issues/index.md]: 谢谢，全部修改完成

**Review by kebe7jun — APPROVED** [review]
LGTM

**Pull request merged**